backend/lightning/lightning.go

```diff
@@ -155,34 +165,57 @@ func (lightning *Lightning) connect() {
 		// support multiple accounts, for future extensions.
 		account := lightningConfig.Accounts[0]
 
-		// TODO: this seed should be determined from the account/device.
 		seed, err := breez_sdk.MnemonicToSeed(account.Mnemonic)
+		if err != nil {
+			lightning.log.WithError(err).Error("BreezSDK: MnemonicToSeed failed")
+			return
+		}
+
+		deviceKey, err := fetchAuthData(lightning.httpClient, greenLightKeyUrl)
+		if err != nil {
+			lightning.log.WithError(err).Error("Greenlight key fetch failed")
+			return
+		}
 
+		deviceCert, err := fetchAuthData(lightning.httpClient, greenLightCertUrl)
 		if err != nil {
-			lightning.log.WithError(err).Warn("BreezSDK: MnemonicToSeed failed")
+			lightning.log.WithError(err).Error("Greenlight cert fetch failed")
 			return
 		}
 
+		greenlightCredentials := breez_sdk.GreenlightCredentials{
+			DeviceKey:  *deviceKey,
+			DeviceCert: *deviceCert,
+		}
+
 		nodeConfig := breez_sdk.NodeConfigGreenlight{
 			Config: breez_sdk.GreenlightNodeConfig{
-				PartnerCredentials: nil,
+				PartnerCredentials: &greenlightCredentials,
 				InviteCode:         nil,
 			},
 		}
 
 		workingDir := path.Join(lightning.cacheDirectoryPath, accountBreezFolder(account.Code))
 
 		if err := os.MkdirAll(workingDir, 0700); err != nil {
-			lightning.log.WithError(err).Warn("Error creating working directory")
+			lightning.log.WithError(err).Error("Error creating working directory")
+			return
+		}
+
+		breezApiKey, err := fetchAuthData(lightning.httpClient, breezApiKeyUrl)
+		if err != nil {
+			lightning.log.WithError(err).Error("Breez api key fetch failed")
 			return
 		}
 
-		config := breez_sdk.DefaultConfig(breez_sdk.EnvironmentTypeProduction, "", nodeConfig)
+		// fetched key could have an unwanted newline, we'll just trim it for safety.
+		trimmedKey := strings.TrimSuffix(string(*breezApiKey), "\n")
```

Comment:
imo just call `strings.TrimSpace()`, it does not hurt and will prevent failure if some other whitespace is accidentally present